code/game/objects/items/devices/scanners.dm

```diff
@@ -28,9 +28,19 @@ MASS SPECTROMETER
 		START_PROCESSING(SSobj, src)
 
 /obj/item/device/t_scanner/proc/flick_sonar(obj/pipe)
-	var/image/I = image('icons/effects/effects.dmi', pipe, "blip", pipe.layer+1)
-	I.alpha = 128
-	flick_overlay_view(I, pipe, 8)
+	if(ismob(loc))
+		var/mob/M = loc
+		var/image/I = image(pipe.icon, get_turf(pipe), pipe.icon_state, pipe.layer+1, pipe.dir)
+		I.alpha = 128
+		I.color = pipe.color
+		if(M.client)
+			flick_overlay(I, list(M.client), 8)
+			for(var/C in pipe.overlays)
+				var/image/pipe_overlay = C
+				var/image/J = image(pipe_overlay.icon, get_turf(pipe), pipe_overlay.icon_state, pipe.layer+1, pipe_overlay.dir)
+				J.alpha = 128
+				J.color = pipe.color
+				flick_overlay(J, list(M.client), 8)

```

Comment:
@xxalpha The appearance way of doing it will be about the same cost, it's still 1 appearance churn, if @XDTM does it right:

Old:
```
var/image/I = blah(pipe)
I.alpha = 128 //1 Churn!
show_that_dang_image(I)
```
What I expect you'll do:
```
var/image/I = new/image()
I.appearance = pipe //1 Churn!
I.alpha = 128 //Oh dear 2 Churns!
show_that_dang_image(I)
```
What you should do:
```
var/image/I = new/image()
var/mutable_appearance/MA = new(pipe)
MA.alpha = 128
I.appearance = MA //1 Churn!
show_that_dang_image(I)
```

Personally, Appearance Churn mitigation is more important than the cost of instantiating one object (the ```mutable_appearance```, which are cheap anyway (they're ```/image``` subtypes iirc))